Sources/PackageCollections/Providers/JSONPackageCollectionProvider.swift

```diff
@@ -191,97 +191,101 @@ struct JSONPackageCollectionProvider: PackageCollectionProvider {
+                             location: package.url.absoluteString,
+                             summary: package.summary,
+                             keywords: package.keywords,
+                             versions: versions,
+                             watchersCount: nil,
+                             readmeURL: package.readmeURL,
+                             license: package.license.flatMap { Model.License(from: $0) },
+                             authors: nil,
+                             languages: nil)
             }
 
-            return .init(identity: .init(url: package.url),
-                         location: package.url.absoluteString,
-                         summary: package.summary,
-                         keywords: package.keywords,
-                         versions: versions,
-                         watchersCount: nil,
-                         readmeURL: package.readmeURL,
-                         license: package.license.flatMap { Model.License(from: $0) },
-                         authors: nil,
-                         languages: nil)
-        }
+            if !serializationOkay {
+                self.observabilityScope.emit(warning: "Some of the information from \(collection.name) could not be deserialized correctly, likely due to invalid format. Contact the collection's author (\(collection.generatedBy?.name ?? "n/a")) to address this issue.")
+            }
 
-        if !serializationOkay {
-            self.observabilityScope.emit(warning: "Some of the information from \(collection.name) could not be deserialized correctly, likely due to invalid format. Contact the collection's author (\(collection.generatedBy?.name ?? "n/a")) to address this issue.")
+            return .success(.init(source: source,
+                                  name: collection.name,
+                                  overview: collection.overview,
+                                  keywords: collection.keywords,
+                                  packages: packages,
+                                  createdAt: collection.generatedAt,
+                                  createdBy: collection.generatedBy.flatMap { Model.Collection.Author(name: $0.name) },
+                                  signature: signature,
+                                  lastProcessedAt: Date()))
+        } catch {
+            return .failure(error)
         }
-
-        return .success(.init(source: source,
-                              name: collection.name,
-                              overview: collection.overview,
-                              keywords: collection.keywords,
-                              packages: packages,
-                              createdAt: collection.generatedAt,
-                              createdBy: collection.generatedBy.flatMap { Model.Collection.Author(name: $0.name) },
-                              signature: signature,
-                              lastProcessedAt: Date()))
```

Comment:
diff is terrible here, but its mostly formatting as the code moved into a `try` block